backend/apps/guests/services.py

```
@@ -196,6 +196,38 @@ def create_guest_pass_for_resident(
+
+    guest_pass.status = GuestPass.Status.CANCELLED
+    guest_pass.cancelled_at = timezone.now()
+    guest_pass.save(update_fields=["status", "cancelled_at", "updated_at"])
+    return guest_pass
+
+
+def revoke_guest_pass_admin(pass_id: int, residence) -> GuestPass:
+    try:
+        guest_pass = GuestPass.objects.get(id=pass_id, residence=residence)
+    except GuestPass.DoesNotExist:
+        raise ValidationError({"detail": "Pase no encontrado."})
```

Ruff (0.15.7)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)